### PR TITLE
Improve i18n reimbursements

### PR DESCRIPTION
**Pull request by Murph33**
Improve the reimbursements views by using I18n model name translation and attribute translations where logical.  Also added in a route to the reimbursements index so it's actually reachable (though no link exists.)

 No links exist to the reimbursement show pages either it appears but this is getting out of scope.

This is part of an ongoing effort to improve the I18n usage in solidus as discussed in #735.

### CODE REVIEW

File: backend/app/views/spree/admin/reimbursements/edit.html.erb [hunk 1]
```diff
@@ -16,11 +16,11 @@
     <table class="index reimbursement-return-items">
       <thead>
         <tr>
-          <th><%= Spree.t(:product) %></th>
+          <th><%= Spree::Product.model_name.human %></th>
           <th><%= Spree.t(:preferred_reimbursement_type) %></th>
           <th><%= Spree.t(:reimbursement_type_override) %></th>
```
Comment (cbrunsdon) [inline backend/app/views/spree/admin/reimbursements/edit.html.erb]: Should these be fixed here too? Or are they going to be a separate PR?

**Comment by mamhoff** [comment]
Can you rebase this PR on current master?

File: backend/app/views/spree/admin/reimbursements/index.html.erb [hunk 2]
```diff
@@ -27,7 +27,6 @@
         <td><%= reimbursement.total %></td>
         <td><%= reimbursement.reimbursement_status %></td>
         <td><%= pretty_time(reimbursement.created_at) %></td>
-        <td></td>
```
Comment (mamhoff) [inline backend/app/views/spree/admin/reimbursements/index.html.erb]: So there is no actions. Should then the header `td` not also be deleted?

File: backend/app/views/spree/admin/reimbursements/show.html.erb [hunk 3]
```diff
@@ -54,12 +54,12 @@
 </fieldset>
 
 <fieldset class="no-border-bottom">
-  <legend align='center'><%= Spree.t(:refunds) %></legend>
+  <legend align='center'><%= Spree::Refund.model_name.human %></legend>
   <table class="index reimbursement-refunds">
     <thead data-hook="customer_return_header">
       <tr>
         <th><%= Spree.t(:description) %></th>
```
Comment (mamhoff) [inline backend/app/views/spree/admin/reimbursements/show.html.erb]: This can also be done using `human_attribute_name`, even though it isn't a database column. It's also kinda special, because it isn't really a description but the payment method's name, so I think it merits having its own translation.

File: backend/app/views/spree/admin/reimbursements/edit.html.erb [hunk 4]
```diff
@@ -16,12 +16,12 @@
...
-          <th><%= Spree.t(:preferred_reimbursement_type) %></th>
-          <th><%= Spree.t(:reimbursement_type_override) %></th>
-          <th><%= Spree.t(:pre_tax_refund_amount) %></th>
-          <th><%= Spree.t(:total) %></th>
-          <th><%= Spree.t(:exchange_for) %></th>
+          <th><%= Spree::Product.model_name.human %></th>
+          <th><%= Spree::ReturnItem.human_attribute_name(:preferred_reimbursement_type_id) %></th>
+          <th><%= Spree::ReturnItem.human_attribute_name(:override_reimbursement_type_id) %></th>
+          <th><%= Spree::ReturnItem.human_attribute_name(:amount) %></th>
+          <th><%= Spree::Reimbursement.human_attribute_name(:total) %></th>
```
Comment (mamhoff) [inline backend/app/views/spree/admin/reimbursements/edit.html.erb]: This is the `total` field on `Spree::ReturnItem`, why do we use the translation from `Spree::Reimbursement` here?

Comment (Murph33) [inline backend/app/views/spree/admin/reimbursements/edit.html.erb]: `Spree::ReturnItem` doesn't have a `total` column in the DB so I went with reimbursement to maintain the wording in my efforts to not add attributenames when there is no column.  In hindsight that was overly stringent.

**Comment by mamhoff** [comment]
:+1:

**Comment by mamhoff** [comment]
Would you mind rebasing again?

**Comment by jhawthorn** [comment]
:+1:
